aioxmpp/bookmarks/service.py

```diff
@@ -21,31 +21,66 @@
 ########################################################################
 import asyncio
 
-import aioxmpp
 import aioxmpp.callbacks as callbacks
 import aioxmpp.service as service
-import aioxmpp.disco as disco
-import aioxmpp.pubsub as pubsub
 import aioxmpp.private_xml as private_xml
 
 from . import xso as bookmark_xso
 
 
 # TODO: use private storage in pubsub where available.
 # TODO: sync bookmarks between pubsub and private xml storage
+# TODO: do we need merge-capabilities to reconcile the bookmarks
+# from different sources (local bookmark storage, pubsub, private xml
+# storage)
 class BookmarkClient(service.Service):
     """
     Supports retrieval and storage of bookmarks on the server.
     It currently only supports :xep:`Private XML Storage <49>` as
     backend.
 
+    There is the general rule *never* to modify the bookmark instances
+    retrieved from this class (either by :meth:`get_bookmarks` or as
+    an argument to one of the signals). If you need to modify a bookmark
+    for use with :meth:`update_bookmark` use :func:`~copy.copy` to create
```

Comment:
For references to non-aioxmpp functions and modules, I generally prefer to have the fully-qualified name listed. That makes things immediately clear and stops the reader from having to interrupt flow to figure out if that is an aioxmpp-specific ``copy`` or the python standard library ``copy``.